exchanges/bitfinex/bitfinex_websocket.go

```diff
@@ -1073,6 +494,617 @@ func (b *Bitfinex) wsHandleData(respRaw []byte) error {
+			return errors.New("unable to type assert position snapshot price")
+		}
+		if position.MarginFunding, ok = positionData[4].(float64); !ok {
+			return errors.New("unable to type assert position snapshot margin funding")
+		}
+		marginFundingType, ok := positionData[5].(float64)
+		if !ok {
+			return errors.New("unable to type assert position snapshot margin funding type")
+		}
+		position.MarginFundingType = int64(marginFundingType)
+		if position.ProfitLoss, ok = positionData[6].(float64); !ok {
+			return errors.New("unable to type assert position snapshot profit loss")
+		}
+		if position.ProfitLossPercent, ok = positionData[7].(float64); !ok {
+			return errors.New("unable to type assert position snapshot profit loss percent")
+		}
+		if position.LiquidationPrice, ok = positionData[8].(float64); !ok {
+			return errors.New("unable to type assert position snapshot liquidation price")
+		}
+		if position.Leverage, ok = positionData[9].(float64); !ok {
+			return errors.New("unable to type assert position snapshot leverage")
+		}
+		snapshot[i] = position
+	}
+	b.Websocket.DataHandler <- snapshot
+	return nil
+}
+
+func (b *Bitfinex) handleWSPositionUpdate(d []interface{}) error {
+	positionData, ok := d[2].([]interface{})
+	if !ok || len(positionData) == 0 {
+		return nil
+	}
+	var position WebsocketPosition
+	if position.Pair, ok = positionData[0].(string); !ok {
+		return errors.New("unable to type assert position pair")
+	}
+	if position.Status, ok = positionData[1].(string); !ok {
+		return errors.New("unable to type assert position status")
+	}
+	if position.Amount, ok = positionData[2].(float64); !ok {
+		return errors.New("unable to type assert position amount")
+	}
+	if position.Price, ok = positionData[3].(float64); !ok {
+		return errors.New("unable to type assert position price")
+	}
+	if position.MarginFunding, ok = positionData[4].(float64); !ok {
+		return errors.New("unable to type assert margin position funding")
+	}
+	marginFundingType, ok := positionData[5].(float64)
+	if !ok {
+		return errors.New("unable to type assert position margin funding type")
+	}
+	position.MarginFundingType = int64(marginFundingType)
+	if position.ProfitLoss, ok = positionData[6].(float64); !ok {
+		return errors.New("unable to type assert position profit loss")
+	}
+	if position.ProfitLossPercent, ok = positionData[7].(float64); !ok {
+		return errors.New("unable to type assert position profit loss percent")
+	}
+	if position.LiquidationPrice, ok = positionData[8].(float64); !ok {
+		return errors.New("unable to type assert position liquidation price")
+	}
+	if position.Leverage, ok = positionData[9].(float64); !ok {
+		return errors.New("unable to type assert position leverage")
+	}
+	b.Websocket.DataHandler <- position
+	return nil
+}
+
+func (b *Bitfinex) handleWSTradeUpdate(d []interface{}, eventType string) error {
+	tradeData, ok := d[2].([]interface{})
+	if !ok || len(tradeData) <= 4 {
+		return nil
```

Comment:
return error

Reply:
I can return error on failed conversion but returning error for tradeData < 4 is more intrusive than this refactor should be doing.
If it's an (existing) issue let's open a separate issue